### PR TITLE
Make Birth Year Required and Add Create Range of Zip Codes functionality

### PR DESCRIPTION
**Pull request by dahnbaez**
To make Birth Year Required is necessary to create the setting "RequiredBirthYear" and set it as true.
I also created a button to Delete a Range of Zip Codes.

### CODE REVIEW

File: CmsWeb/Areas/Setup/Views/MetroZip/Index.cshtml [hunk 1]
```diff
@@ -142,6 +213,27 @@
                 return false;
             });
 
+            $("a.create-range").click(function (ev) {
+                ev.preventDefault();
+                $('#range-modal').modal();
+                $.getJSON('/MetroZips/ResidentCodes/', function (data) {
+                    let dropdown = $('#marginaldd');
+                    dropdown.empty();
+                    dropdown.append('<option selected="true" value="0" disabled>Choose Metro Maginal Code</option>');
```
Comment (brondavies) [inline CmsWeb/Areas/Setup/Views/MetroZip/Index.cshtml]: typo: Ma**r**ginal?

File: CmsWeb/Content/touchpoint/js/metrozip/metro-zip.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
+﻿$('document').ready(function () {
```
Comment (brondavies) [inline CmsWeb/Content/touchpoint/js/metrozip/metro-zip.js]: can be shortened to `$(function () {`

File: CmsWeb/Content/touchpoint/js/metrozip/metro-zip.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            valid = false;
+        }
+        if (parseInt(drStartWithInput.val()) > parseInt(drEndWithInput.val())) {
+            valid = false;
+        }        
+        if (!valid) {
+            delRangeBtn.prop('disabled', true);
+        } else {
+            delRangeBtn.prop('disabled', false);
+        }
```
Comment (brondavies) [inline CmsWeb/Content/touchpoint/js/metrozip/metro-zip.js]: You can reduce this if block to just `delRangeBtn.prop('disabled', !valid);`

File: CmsWeb/Views/Shared/EditorTemplates/Birthday.cshtml [hunk 4]
```diff
@@ -16,3 +19,19 @@
         @Html.ValidationMessageLabelFor(m => m)
     </div>
 </div>
+@if (requiredBirthYear)
```
Comment (brondavies) [inline CmsWeb/Views/Shared/EditorTemplates/Birthday.cshtml]: What is the purpose of this code? I tried entering a birth date on the admin profile and it didn't change the behavior https://pr360.tpsdb.com/Person2/1

Comment (brondavies) [reply CmsWeb/Views/Shared/EditorTemplates/Birthday.cshtml]: OK I checked the .work site again and it doesn't look like the code has been updated there yet.  I'll ping @aternex and test again

File: CmsWeb/Content/touchpoint/js/metrozip/metro-zip.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        if (arStartWithInput.val() == '' || arEndWithInput.val() == '') {
+            valid = false;
+        }
+        if (parseInt(arStartWithInput.val()) > parseInt(arEndWithInput.val())) {
+            valid = false;
+        }
+        if (residentCodeDD.val() == null) {
+            valid = false;
+        }
+        if (!valid) {
```
Comment (brondavies) [inline CmsWeb/Content/touchpoint/js/metrozip/metro-zip.js]: Also can be shortened to `sendRangeBtn.prop('disabled', !valid);`

File: CmsWeb/Areas/Setup/Views/MetroZip/Index.cshtml (hunk 1 above)
Comment (dahnbaez) [reply CmsWeb/Areas/Setup/Views/MetroZip/Index.cshtml]: I think it's better if we change to "Resident Code"

File: CmsWeb/Views/Shared/EditorTemplates/Birthday.cshtml (hunk 4 above)
Comment (dahnbaez) [reply CmsWeb/Views/Shared/EditorTemplates/Birthday.cshtml]: The idea of this is blocking any input interaction and force to the user to use the datepicker.

File: CmsWeb/Areas/Setup/Views/MetroZip/Index.cshtml (hunk 1 above)
Comment (brondavies) [reply CmsWeb/Areas/Setup/Views/MetroZip/Index.cshtml]: sounds good.

File: CmsWeb/Views/Shared/EditorTemplates/Birthday.cshtml (hunk 4 above)
Comment (brondavies) [reply CmsWeb/Views/Shared/EditorTemplates/Birthday.cshtml]: You should probably also see if we can add a server-side validation that will require the year if the setting is turned on.

**Pull request merged**